src/components/HFUI/HFUI.js

```diff
@@ -38,6 +44,29 @@ const HFUI = ({
       shouldShowAOPauseModalState()
     }
   }
+  const closeBestExperienceModal = () => setBestExperienceModalState(false)
+
+  const onSubmitBestExperienceModal = () => {
+    localStorage.setItem(HF_UI_WEB_SHOW_BEST_EXPERIENCE_MODAL, 'false')
+    closeBestExperienceModal()
+  }
+
+  useEffect(() => {
+    const currentWidth = document.documentElement.clientWidth
+    if (isElectronApp || currentWidth >= MIN_SAFE_WIDTH) {
+      return
+    }
+    let needToShowModal = localStorage.getItem(HF_UI_WEB_SHOW_BEST_EXPERIENCE_MODAL)
+    if (_isNil(needToShowModal)) {
+      needToShowModal = true
+    } else {
+      needToShowModal = JSON.parse(needToShowModal)
+    }
+    if (!needToShowModal) {
+      return
+    }
+    setBestExperienceModalState(true)
+  }, [])
```

Comment:
looks like we can move this logic to BestExperienceMessageModal.js, this way relevant logic stays together and we avoid polluting HFUI too much